core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -385,6 +368,141 @@ sealed abstract class MultiSigSigner
+      Future.fromTry(TxBuilderError.WrongSigner)
+    } else {
+      val (_, output, inputIndex, _) =
+        relevantInfo(spendingInfo, unsignedTx)
+
+      (spendingInfoToSatisfy, spendingInfoToSatisfy.redeemScript) match {
+        case (_: P2SHNestedSegwitV0UTXOSpendingInfo,
+              p2wpkh: P2WPKHWitnessSPKV0) =>
+          val witTx = WitnessTransaction.toWitnessTx(unsignedTx)
+          val p2shScriptSig = P2SHScriptSignature(p2wpkh)
+          val oldInput = witTx.inputs(inputIndex.toInt)
+          val updatedInput = TransactionInput(oldInput.previousOutput,
+                                              p2shScriptSig,
+                                              oldInput.sequence)
+
+          val uwtx = {
+            val u = unsignedTx.updateInput(inputIndex.toInt, updatedInput)
+            WitnessTransaction.toWitnessTx(u)
+          }
+
+          val wtxSigComp = WitnessTxSigComponentP2SH(transaction = uwtx,
+                                                     inputIndex = inputIndex,
+                                                     output = output,
+                                                     flags =
+                                                       Policy.standardFlags)
+
+          val hashForSig = TransactionSignatureSerializer.hashForSignature(
+            txSigComponent = wtxSigComp,
+            hashType = spendingInfo.hashType)
+
+          //sign the hash
+          val signature: ECDigitalSignature = {
+            if (isDummySignature) {
+              EmptyDigitalSignature
+            } else {
+              val sig = spendingInfo.signers.head.sign(hashForSig.bytes)
+              //append hash type
+              ECDigitalSignature.fromBytes(
+                sig.bytes.:+(spendingInfo.hashType.byte))
+            }
+          }
+          val p2wpkhWit =
+            P2WPKHWitnessV0(spendingInfo.signers.head.publicKey, signature)
+          val updatedWit = uwtx.updateWitness(inputIndex.toInt, p2wpkhWit)
+
+          Future.successful(
+            P2SHTxSigComponent(updatedWit, inputIndex, output, flags))
+        case (nestedSpendingInfo: P2SHNestedSegwitV0UTXOSpendingInfo,
+              _: P2WSHWitnessSPKV0) =>
+          val outpoint = spendingInfo.outPoint
+          val redeemScript = nestedSpendingInfo.redeemScript
+          val oldInput = unsignedTx.inputs(inputIndex.toInt)
+
+          val input =
+            TransactionInput(outpoint, EmptyScriptSignature, oldInput.sequence)
+
+          val updatedTx =
+            unsignedTx.updateInput(inputIndex.toInt, input)
+
+          val updatedOutput =
+            TransactionOutput(spendingInfo.amount, redeemScript)
+
+          val updatedUTXOInfo = BitcoinUTXOSpendingInfo(
+            outpoint,
+            updatedOutput,
+            spendingInfo.signers,
+            None,
+            spendingInfo.scriptWitnessOpt,
+            spendingInfo.hashType,
+            spendingInfo.conditionalPath)
```

Comment:
nit: I think that p2sh spending info has a field called `nestedSpendingInfo` that should do this for you IIRC